cmake/CubBuildCompilerTargets.cmake

```diff
@@ -90,7 +92,7 @@ function(cub_build_compiler_targets)
       # CMake can't split genexs, so this can't be formatted better :(
       # This is:
       # if (using CUDA and CUDA_COMPILER is NVCC) add -Xcompiler=opt:
-      $<$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>:-Xcompiler=${cxx_option}>
+      $<$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>:${cuda_compile_options} -Xcompiler=${cxx_option}>
```

Comment:
This will specify `-Wreorder` multiple times, once for each `cxx_option`. This should be done outside of the `foreach` loop.